tasks/postgresql.yml

```diff
@@ -26,7 +26,28 @@
 - name: Configure postgresql repository
   package:
     name: "{{ postgresql_yum_repo_url }}"
-  when: ansible_os_family == "RedHat"
+  when: 
+    - ansible_os_family == "RedHat"
+    - ansible_distribution_major_version <= "7"
+
+- block:
+  - name: "Configure postgresql repository key"
+    rpm_key:
+      state: present
+      key:  "{{ postgresql_yum_repo_gpg_key_url }}"
+  
+  - name: Configure postgresql repository
+    package:
+      name: "{{ postgresql_yum_repo_url }}"
+      state: present
+      use: dnf

```

Comment:
I guess you just need to tell Ansible which package manager to use when the default is wrong. e.g. when you want a more exotic one that's just an addition on a system.

```suggestion

```